tests/data/ar/CustomerWithConstructor.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace yiiunit\data\ar;
+
+use yiiunit\framework\db\ActiveRecordTest;
+
+/**
+ * Class Customer
+ *
+ * @property int $id
+ * @property string $name
+ * @property string $email
+ * @property string $address
+ * @property int $status
+ *
+ * @property ProfileWithConstructor $profile
+ */
+class CustomerWithConstructor extends ActiveRecord
+{
+    const STATUS_ACTIVE = 1;
+    const STATUS_INACTIVE = 2;
+
+    public static function tableName()
+    {
+        return 'customer';
+    }
+
+    public function __construct($name, $email, $address)
+    {
+        $this->name = $name;
+        $this->email = $email;
+        $this->address = $address;
+        parent::__construct();
+    }
+
+    public static function instantiate($row = [])
+    {
+        return (new \ReflectionClass(static::className()))->newInstanceWithoutConstructor();
```

Comment:
i think thats up for the developer to know how to handle the constructor and the instantiate.

lets think it the other way around. what if the developer assumes some property, behavior or something got initialized by the constructor and start using it after a `findOne()` for example.